### PR TITLE
Refactor integration test framework.

### PR DESCRIPTION
**Pull request by qinggniq**
### What problem does this PR solve? 
Refactor integration framework.

### What is changed and how it works?
Previous integration test framework has many dup code, using interface to refactor it.

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

### Release note

- No release note

### CODE REVIEW

**Comment by qinggniq** [comment]
@liuzix PTAL

File: integration/integration.go [hunk 1]
```diff
@@ -15,15 +15,14 @@ package main
 
 import (
 	"flag"
+
 	"github.com/pingcap/log"
 	"github.com/pingcap/ticdc/integration/framework"
+	avro2 "github.com/pingcap/ticdc/integration/framework/avro"
```
Comment (liuzix) [inline integration/integration.go:21]: Please rethink the naming

File: integration/tests/case_alter.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+// AlterCase is base impl of test case for alter operation
+type AlterCase struct {
+	framework.Task
+}
+
+// NewAlterCase create a test case which contains alter ddls
+func NewAlterCase(typ ProtocolType) *AlterCase {
+	switch typ {
```
Comment (liuzix) [inline integration/tests/case_alter.go:34]: Please avoid writing this boilerplate so many times

**Review by liuzix — COMMENTED** [review]
Two main issues:
- Duplicate package names could lead to confusion.
- You should let each case constructor take a `SingleTableTask` interface, which should be constructed properly in the main function according to user's specification. Plus, currently to add a new case you need to create at least three files. Please rethink the design.

**Comment by qinggniq** [comment]
@liuzix PTAL.

**Comment by qinggniq** [comment]
@liuzix PTAL

**Comment by qinggniq** [comment]
@liuzix PTAL

**Pull request merged**